### PR TITLE
Use ES6 Classes

### PR DESCRIPTION
**Pull request by camillobruni**

### CODE REVIEW

**Comment by camillobruni** [comment]
Given that this might be hard to diff, I tried only restructure code at the function-level and leave the rest untouched.

**Comment by camillobruni** [comment]
Note 2: I will format the code in follow-up CL to make this review easier.

File: resources/benchmark-runner.js [hunk 1]
```diff
@@ -1,15 +1,65 @@
-function BenchmarkTestStep(testName, testFunction) {
+class BenchmarkTestStep {
+constructor(testName, testFunction) {
     this.name = testName;
     this.run = testFunction;
 }
+}
+
+class BenchmarkState {
```
Comment (rniwa) [inline resources/benchmark-runner.js:8]: Can we not change the order in which BenchmarkRunner and BenchmarkState are declared / defined?

File: resources/benchmark-runner.js [hunk 2]
```diff
@@ -1,15 +1,65 @@
...
+    this._testIndex = 0;
+    this.next();
+}
+
+currentSuite() {
+    return this._suites[this._suiteIndex];
+}
+
+currentTest() {
+    let suite = this.currentSuite();
```
Comment (rniwa) [inline resources/benchmark-runner.js:21]: We can use const here.

File: resources/main.js [hunk 3]
```diff
@@ -1,33 +1,41 @@
-window.benchmarkClient = {
-    displayUnit: 'runs/min',
-    iterationCount: 10,
-    stepCount: null,
-    suitesCount: null,
-    _measuredValuesList: [],
-    _finishedTestCount: 0,
-    _progressCompleted: null,
+// TODO(camillobruni): Add base class
```
Comment (rniwa) [inline resources/main.js:1]: Please use FIXMEs instead of TODOs.

**Review by rniwa — APPROVED** [review]

**Review by bgrins — APPROVED** [review]

**Pull request merged**